test/check/events.go

```diff
@@ -209,3 +209,16 @@ func (w *ObjectEventWatcher) WaitNotFor(stopChan chan struct{}, eventType EventT
 	}, fmt.Sprintf("not happen event type %s, reason = %s", string(eventType), reflect.ValueOf(reason).String()))
 	return
 }
+
+func (w *ObjectEventWatcher) WaitNotForType(stopChan chan struct{}, eventType EventType) (e *corev1.Event) {
+	w.dontFailOnMissingEvent = true
+	w.Watch(stopChan, func(event *corev1.Event) bool {
+		if event.Type == string(eventType) {
+			e = event
+			Fail(fmt.Sprintf("Did not expect %s. reason is %s", string(eventType), event.Reason), 1)
```

Comment:
Do we need `defer GinkgoRecover` inside the func ? since this run at a coroutine.

Reply:
I think we already have `defer GinkgoRecover` inside the [Watch() function](https://github.com/kubevirt/cluster-network-addons-operator/blob/master/test/check/events.go#L163) where the goroutine is actually running

Reply:
Then we good.